Sources/CacheAdvance/FileHandleExtensions.swift

```diff
@@ -0,0 +1,160 @@
+            }
+            return nil
+
+        case .invalidFormat:
+            throw CacheAdvanceReadError.fileCorrupted
+        }
+    }
+
+    /// Seeks just ahead of the newest message in the file.
+    /// - Parameter cacheOverwritesOldMessages: When `true`,  the cache encodes a pointer to the oldest message after the newest message marker.
+    func seekToEndOfNewestMessage(cacheOverwritesOldMessages: Bool) throws {
+        while try seekToNextMessage(shouldSeekToOldestMessageIfFound: false, cacheOverwritesOldMessages: cacheOverwritesOldMessages) {}
+    }
+
+    /// Seeks to the beginning of the oldest message in the file.
+    /// - Parameter cacheOverwritesOldMessages: When `true`,  the cache encodes a pointer to the oldest message after the newest message marker.
+    func seekToBeginningOfOldestMessage(cacheOverwritesOldMessages: Bool) throws {
+        if cacheOverwritesOldMessages {
+            while try seekToNextMessage(shouldSeekToOldestMessageIfFound: true, cacheOverwritesOldMessages: true) {}
+        } else {
+            // The oldest message is always at the beginning of the cache.
+            try seek(toOffset: 0)
+        }
+    }
+
+    /// Seeks to the next message. Returns `true` when the span skipped represented a message.
+    /// When `false` is returned, it signifies that the last message marker was passed.
+    ///
+    /// - Parameters:
+    ///   - shouldSeekToOldestMessageIfFound: When `true`, the file handle will seek to the oldest message if the last message marker is pased.
+    ///   - cacheOverwritesOldMessages: When `true`,  the cache encodes a pointer to the oldest message after the newest message marker.
+    @discardableResult
+    func seekToNextMessage(shouldSeekToOldestMessageIfFound: Bool, cacheOverwritesOldMessages: Bool) throws -> Bool {
+        let startingOffset = offsetInFile
+        switch try nextEncodedMessageSpan(cacheOverwritesOldMessages: cacheOverwritesOldMessages) {
+        case let .endOfNewestMessageMarker(offsetOfNextMessage):
+            // We found the last message!
+            if shouldSeekToOldestMessageIfFound {
+                // Seek to it.
+                try seek(toOffset: offsetOfNextMessage)
+            } else {
+                // Seek back to where we started.
+                try seek(toOffset: startingOffset)
+            }
+            return false
+
+        case .emptyRead:
+            // The remaining file is empty.
+            if shouldSeekToOldestMessageIfFound {
+                // The oldest message is at the beginning of the file.
+                try seek(toOffset: 0)
+            } else {
+                // Seek back to where we started.
+                try seek(toOffset: startingOffset)
+            }
+            return false
+
+        case let .span(messageLength):
+            // There's a valid message here. Seek ahead of it.
+            try seek(toOffset: offsetInFile + UInt64(messageLength))
```

Comment:
Should we be using one of our type aliases (e.g. `Bytes`) instead of `UInt64`?

Reply:
since `seek` takes a `UInt64`, it seemed more semantic to use that type here.

Imagine a world where we decide to only support a 4GB file and change `Bytes` to be a typealias for `UInt32. In that world, we wouldn't want this line to fail.